src/tokens/wrappers/clawback/IClawback.sol

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.19;
+
+interface IClawbackFunctions {
+    enum TokenType {
+        ERC20,
+        ERC721,
+        ERC1155
+    }
+
+    struct Template {
+        address admin;
+        uint96 duration;
+        bool destructionOnly;
+        bool transferOpen;
+    }
+
+    struct TokenDetails {
```

Comment:
Lets use this struct instead:

```
    struct TokenDetails {
        TokenType tokenType;
        uint32 templateId;
        uint56 lockedAt;
        address tokenAddr;
        uint256 tokenId; // 0 for ERC20
    }
```
It packs into 2 slots instead of 3, using `uint32` for `templateId` means that attacking the contract by creating tons of fake templates becomes pretty much impossible. `uint56` is more than enough for an unix time.